pyre/check.py

```
@@ -1107,6 +1107,52 @@ def synth_perf_gate(path):
+    ratio = None
+    with open(path, encoding="utf-8") as source:
+        for _ in range(20):
+            line = source.readline()
+            if not line:
+                break
+            if not line.startswith(prefix) or ratio is not None:
+                continue
+            try:
+                ratio = float(line[len(prefix):].strip())
+            except ValueError as e:
+                raise ValueError(f"invalid wasm ratio gate in {path}: {line.strip()}") from e
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)